lib/src/model/swipe_rate_per_threshold.dart

```diff
@@ -10,41 +10,102 @@ class _SwipeRatePerThreshold {
+      case SwipeDirection.up:
+        return rate.clamp(double.negativeInfinity, 0);
+      case SwipeDirection.down:
+        return rate.clamp(0, double.infinity);
+    }
+  }
+}
+
 extension _SwipableStackPositionX on _SwipableStackPosition {
-  _SwipeRatePerThreshold swipeDirectionRate({
+  _SwipeRatePerThreshold? swipeDirectionRate({
     required BoxConstraints constraints,
     required double horizontalSwipeThreshold,
     required double verticalSwipeThreshold,
+    required Set<SwipeDirection> detectableDirections,
   }) {
-    final horizontalRate = (difference.dx.abs() / constraints.maxWidth) /
-        (horizontalSwipeThreshold / 2);
-    final verticalRate = (difference.dy.abs() / constraints.maxHeight) /
-        (verticalSwipeThreshold / 2);
-    final horizontalRateGreater = horizontalRate >= verticalRate;
-    if (horizontalRateGreater) {
+    final horizontalRate =
+        (difference.dx / constraints.maxWidth) / (horizontalSwipeThreshold / 2);
+    final verticalRate =
+        (difference.dy / constraints.maxHeight) / (verticalSwipeThreshold / 2);
+
+    _SwipeRatePerThreshold? horizontalRatePerThreshold() {
+      final filteredDirs = [SwipeDirection.left, SwipeDirection.right]
+          .where(detectableDirections.contains);
+      if (filteredDirs.isEmpty) {
+        return null;
+      }
+      if (filteredDirs.length == 2) {
+        return _SwipeRatePerThreshold(
+          direction:
+              difference.dx >= 0 ? SwipeDirection.right : SwipeDirection.left,
+          rate: horizontalRate.abs(),
+        );
+      }
+      final dir = filteredDirs.first;
       return _SwipeRatePerThreshold(
-        direction:
-            difference.dx >= 0 ? SwipeDirection.right : SwipeDirection.left,
-        rate: horizontalRate,
+        direction: dir,
+        rate: dir.clampSwipeRate(horizontalRate).abs(),
       );
-    } else {
+    }
+
+    _SwipeRatePerThreshold? verticalRatePerThreshold() {
+      final filteredDirs = [SwipeDirection.up, SwipeDirection.down]
+          .where(detectableDirections.contains);
+      if (filteredDirs.isEmpty) {
+        return null;
+      }
+      if (filteredDirs.length == 2) {
+        return _SwipeRatePerThreshold(
+          direction:
+              difference.dy >= 0 ? SwipeDirection.down : SwipeDirection.up,
+          rate: verticalRate.abs(),
+        );
+      }
+      final dir = filteredDirs.first;
       return _SwipeRatePerThreshold(
-        direction: difference.dy >= 0 ? SwipeDirection.down : SwipeDirection.up,
-        rate: verticalRate,
+        direction: dir,
+        rate: dir.clampSwipeRate(verticalRate).abs(),
       );
     }
+
+    final rateList = [
+      verticalRatePerThreshold(),
+      horizontalRatePerThreshold(),
+    ].whereType<_SwipeRatePerThreshold>().toList();
+    if (rateList.isEmpty) {
+      return null;
+    }
+    return (rateList
+          ..sort(
+            (a, b) => a.rate > b.rate ? -1 : 1,
+          ))
+        .first;
```

Comment:
```suggestion
    return rateList.length == 1 || rateList[0].rate > rateList[1] ? rateList[0] : rateList[1];
```
I don't know if the sort function is a little overkill at this point. But it is of course more elegant and flexible than the brute force method.

Reply:
Thanks for your suggestion. This code is more understandable! 🙏